docs/inputs.md

```diff
@@ -13,11 +13,19 @@ inputs:
     url: github:cachix/pre-commit-hooks.nix
 ```
 
-Input names like ``nixpkgs`` and ``pre-commit-hooks`` are identifiers for what
- is passsed in the first line of the function:
+The dependencies you mention as `inputs` are passsed as an argument to the function.
 
 ```nix title="devenv.nix"
-{ pkgs, lib, nixpkgs, pre-commit-hooks, config, ... }:
+{ inputs, ... }:
+
+{
```

Comment:
Let's show an example how to reference for example pre-commit-hooks from the inputs

Reply:
Getting

```
error: The option `inputs' does not exist. Definition values:
       - In `/home/sayanarijit/Documents/GitHub/devenv/devenv.nix':
           {
             pre-commit = {
               hooks = {
                 nixpkgs-fmt = {
                   enable = true;
           ...

       … while evaluating the attribute 'config'

       at https://github.com/NixOS/nixpkgs/blob/69335c46c48a73f291d5c6f332fb9fe8b8e22b30/lib/modules.nix:330:9:

          329|         options = checked options;
          330|         config = checked (removeAttrs config [ "_module" ]);
             |         ^
          331|         _module = checked (config._module);
```

Reply:
You mean something like https://github.com/cachix/devenv/pull/112/files#diff-cef540503c67fa4a36d1b9fe95a62f979c4b8b2bc43a6e023e250b7154341afcR3-R23?

It's not working btw.

Reply:
That's probably not the best example, maybe we can use an example of using git from a stable version of Nix:

```yaml
inputs:
  nixpkgs-stable:
    url: github:NixOS/nixpkgs/nixpkgs-22.05
```

And then:

```nix
{ inputs, system, ... }:

let 
  nixpkgs-stable = import inputs.nixpkgs-stable { inherit system; };
in {
  packages = [ pkgs-stable.git ];
}
```

Reply:
`github:NixOS/nixpkgs/nixpkgs-22.05` doesn't resolve. I think you meant `github:NixOS/nixpkgs/nixos-22.05`. Also, `system` isn't a standalone argument. `pkgs.system` works.

This brings another scope for improvement:

How about we could declare it like:

```yaml
inputs:
  nixpkgs-unstable:  # Defaults to `url: github:NixOS/nixpkgs/nixpkgs-unstable`
  nixos-22.05:  # Defaults to `url: github:NixOS/nixpkgs/nixos-22.05`
```

Reply:
Once we introduce registry this is quite possible

Reply:
I meant how about letting `url: github:NixOS/nixpkgs` be the default registry? Ultil we have a separate registry field, we have to be specific when adding non github:NixOS/nixpkgs deps.